### ISSUE TITLE
Error while upgrade from 2.0.7 to 2.1

### ISSUE THREAD

**Issue by klict** [magento/magento2#5229]
We have tried to upgrade Magento twice now but both times we got an Error
## Steps to reproduce
1. upgrading to 2.1.0 from Setup Wizard
## Expected result
1. Magento 2.1 installed
## Actual result
1. we got an error
## additional information

server runs on nginx and php 7.0.7

[error1.txt](https://github.com/magento/magento2/files/331832/error1.txt)
[error2.txt](https://github.com/magento/magento2/files/331831/error2.txt)

**Comment by developer-lindner** [comment]
Seems to be the same issue as i noted here https://github.com/magento/magento2/issues/5223#issuecomment-228555734

**Comment by PierAlex** [comment]
Same situation, Upgrading from 2.07 to 2.10 via web setup wizard. nginx and php 7.07 (on plesk).
I had the same problem with a different error, My error was refering to the 'attribute set ID' instead of the area code.
After restoring backups, at the third time, i got the success for the installation also if the setup still display the error and the admin is not running:

![error](https://cloud.githubusercontent.com/assets/17165819/16360856/85bb0bf4-3b78-11e6-8714-8637dae515eb.JPG)

and the log:

...
...
...
Module 'Magento_CatalogWidget':
Module 'Magento_Wishlist':
Please re-run Magento compile command

[2016-06-25 22:44:45 UTC] Job "setup:upgrade {"command":"setup:upgrade"}" has been successfully completed
[2016-06-25 22:44:45 UTC] Job "setup:static:regenerate []" has started

[2016-06-25 22:44:45 UTC] Cleaning generated files...
[2016-06-25 22:44:45 UTC] Clearing cache...
[2016-06-25 22:44:45 UTC] Cleaning static view files
[2016-06-25 22:44:45 UTC] Job "setup:static:regenerate []" has been successfully completed
[2016-06-25 22:45:03 UTC] Job "maintenance_mode {"enable":false}" has been started
[2016-06-25 22:45:03 UTC] Magento maintenance mode is disabled.
[2016-06-25 22:45:03 UTC] Job "maintenance_mode {"enable":false}" has successfully completed

Error after login to admin:
[error admin.txt](https://github.com/magento/magento2/files/333465/error.admin.txt)

**Comment by redelschaap** [comment]
@klict is on vacation, I am his colleague.

A strange detail is that after using the rollback button and restoring a back up of 2.0.7 (both files and database), Magento kept saying it couldn't load a couple of (2.0.1) classes. `Magento\Framework\App\Response\HeaderManager` was one of them. We had to restore a snapshot of our server to get Magento running again.

**Comment by andimov** [comment]
@PierAlex 
Check permissions http://devdocs.magento.com/guides/v2.0/config-guide/prod/prod_file-sys-perms.html

**Comment by andimov** [comment]
@redelschaap 
Thank you for reporting. 
Have you tried to upgrade to 2.1 throught command line?

**Comment by redelschaap** [comment]
@andimov No, haven't tried that yet. Do you have the commands we need to use?

**Comment by developer-lindner** [comment]
I've upgraded through CLI.
@redelschaap composer update

**Comment by andimov** [comment]
@redelschaap 
Here you can find entire info for [upgrading throught CLI](http://devdocs.magento.com/guides/v2.0/comp-mgr/cli/cli-upgrade.html). 
Commands list you will find in section **Upgrade using the command line**

**Comment by lucasalfaro** [comment]
Hi, I did upgrade through the composer. I also get

---

Module 'Magento_Catalog':
Upgrading data.. 

  [Magento\Framework\Exception\LocalizedException]  
  Wrong attribute set ID                            

setup:upgrade [--keep-generated] [--magento-init-params="..."]

---

Not sure how to proceed.

**Comment by PierAlex** [comment]
@andimov  No problems with permissions. it was a missing '@' before is_dir at line 914 of the file /vendor/composer/composer/src/Composer/Util/RemoteFilesystem.php

https://github.com/magento/magento2/issues/4359

@lucasalfaro  Wrong attribute set ID. It happens when the 'Default' Attribute set is renamed. If you rename back it in 'Default', the installation goes on.

**Comment by lucasalfaro** [comment]
Excellent, thanks for the answer. Could you please clarify? I didn´t touch the code, therefore I am assuming it is product attribute on the admin, is that correct?

**Comment by PierAlex** [comment]
Yes, in the admin, Stores -> Attribute set.

**Comment by andimov** [comment]
@PierAlex 
Thank you for solution!
@redelschaap 
Is that solution works for you?

**Comment by ghost** [comment]
I am experiencing the same issue here just like #4359 . Manually adding the '@' before is_dir at line 914 of the file /vendor/composer/composer/src/Composer/Util/RemoteFilesystem.php did make it work. Without this modification I was unable to update or access the admin.

**Comment by oserediuk** [comment]
@klict Do you have any updates?

**Comment by bartoszkubicki** [comment]
same issue here - I am trying to remove sets programatically and I call in loop
 `$eavSetup->removeAttributeSet($this->_entityTypeId, $set['name']);` and I get:
`[Magento\Framework\Exception\LocalizedException]  
  Wrong attribute set ID`
assuming values return are real attribute sets values.
`$this->_entityTypeId` equals `ProductAttributeInterface::ENTITY_TYPE_CODE` which equals 4.

**Comment by andimov** [comment]
@klict
According to [contributor guide](http://devdocs.magento.com/guides/v2.1/contributor-guide/contributing.html), tickets without response for two weeks should be closed.
Please feel free to reopen if it's needed.

@anyone 
Please create a separate ticket for other issues and provide detailed steps according to the template. This will help us to process issues. I'm closing this one.

**Comment by redelschaap** [comment]
My apologies for the late response, we had some other projects to complete first.

I've successfully upgraded the installation to version 2.1 after debugging with `bin/magento -vvv setup:upgrade` and temporarily adjusting (core) files. I noted the exceptions we got and how we resolved them. We also applied patch MDVA-532 and made the change @PierAlex suggested, just in case.

```
Exception(s):
    Magento\Framework\Exception\SessionException:
        Area code not set: Area code must be set before starting a session.
    Magento\Framework\Exception\LocalizedException:
        Area code is not set
File:
    /var/www/domains/premium-hookahs.magedev.webwhales.nl/domain/public_html/app/code/J2t/Rewardpoints/Observer/ProcessSaveModel.php:21
Resolved by:
    Temporarily adjusting 3rd party extension (J2t Rewardpoints) to prevent getting Helper class through the object manager


Exception(s):
    InvalidArgumentException:
        <store view code> view does not exist
Resolved by:
    Running setup:upgrade again


Exception(s):
    Magento\Framework\Exception\LocalizedException:
        The page URL key contains capital letters or disallowed symbols.
File:
    /var/www/domains/premium-hookahs.magedev.webwhales.nl/domain/public_html/vendor/magento/module-cms/Model/ResourceModel/Page.php:119
Resolved by:
    Temporarily adjusting 
        if (!$this->isValidPageIdentifier($object)) {
    to
        if (false&&!$this->isValidPageIdentifier($object)) {
    in
        Magento\Cms\Model\ResourceModel\Page->_beforeSave()


Exception(s):
    Zend_Db_Statement_Exception: SQLSTATE[23000]:
        Integrity constraint violation: 1048 Column 'meta_robots' cannot be null, query was: INSERT INTO `cms_page` (`page_id`, `title`, `page_layout`, `meta_keywords`, `meta_description`, `meta_robots`, `identifier`, `content_heading`, `content`, `layout_update_xml`, `custom_theme`, `custom_root_template`, `custom_layout_update_xml`, `custom_theme_from`, `custom_theme_to`, `meta_title`) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)
    PDOException: SQLSTATE[23000]:
        Integrity constraint violation: 1048 Column 'meta_robots' cannot be null
Resolved by:
    Adjusting `cms_page.meta_robots` to accept NULL values, column was added by 3rd party extension (MageWorx SeoBase)
```

So the exceptions were caused by both Magento core modules and 3rd party extensions. Everything works fine now, altough the link to the web updater disappeared in the admin menu.

**Comment by lfritsche** [comment]
@andimov please reopen ticket, because renaming default attribute_set from "Default" to something else should not result in this exception.

I manually had to rename a Attribute Set to "Default" to proceed the bin/magento setup:upgrade process according to https://github.com/magento/magento2/issues/5229#issuecomment-229086685. Isn't there a possibility to allow renaming?

Thank you!

**Comment by magento-engcom-team** [comment]
@klict, thank you for your report.
The issue is already fixed in develop branch, 2.2.1
